File: java/dnd-character/src/main/java/Dice.java
import java.util.Random;

class Dice {
    private int max;
    private Random random = new Random();

    Dice(int max) {
        this.max = max;
    }

    DiceResult roll(int times) {
        DiceResult result = new DiceResult();

        for (int i = 0; i < times; i++) {
            int value = random.nextInt(max) + 1;
            result.add(value);
        }

        return result;
    }
}

File: javascript/dnd-character/dice.js
import { DiceResult } from './dice-result.js';

function random(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

export class Dice {
    constructor(min, max) {
        this.min = min;
        this.max = max;
    }

    roll(n) {
        const values = new Array(n).fill().map(() => random(this.min, this.max));
        return new DiceResult(values);
    }
}
